pkg/netpol/eval/resources.go

```diff
@@ -52,26 +53,67 @@ func NewPolicyEngine() *PolicyEngine {
 		podOwnersToRepresentativePodMap: make(map[string]map[string]*k8s.Pod),
 		cache:                           newEvalCache(),
 		exposureAnalysisFlag:            false,
+		representativePeersMap:          nil, // initialized only for exposure analysis
 	}
 }
 
 func NewPolicyEngineWithObjects(objects []parser.K8sObject) (*PolicyEngine, error) {
 	pe := NewPolicyEngine()
-	err := pe.AddObjects(objects)
+	err := pe.AddObjectsByKind(objects)
 	return pe, err
 }
 
 // NewPolicyEngineWithOptions returns a new policy engine with an empty state but updating the exposure analysis flag,
-// TBD: currently exposure-analysis is the only option supported by policy-engine, so no need for options param,
-// should have the exposureFlag or just assign to true in the func?
+// TBD: currently exposure-analysis is the only option supported by policy-engine, so no need for options list param
 func NewPolicyEngineWithOptions(exposureFlag bool) *PolicyEngine {
 	pe := NewPolicyEngine()
 	pe.exposureAnalysisFlag = exposureFlag
+	if exposureFlag {
+		pe.representativePeersMap = make(map[string]*k8s.RepresentativePeer)
+	}
 	return pe
 }
 
-// AddObjects adds k8s objects from parsed resources to the policy engine
+// AddObjects adds k8s objects to the policy engine: first adds network-policies and then other objects
+// called only for exposure analysis; otherwise does nothing
 func (pe *PolicyEngine) AddObjects(objects []parser.K8sObject) error {
+	if !pe.exposureAnalysisFlag { // should not be true ever
+		return nil
+	}
+	policies, nonPolicies := splitPoliciesAndOtherObjects(objects)
+	err := pe.addPolicies(policies)
+	if err != nil {
+		return err
+	}
+	err = pe.AddObjectsByKind(nonPolicies)
+	return err
+}
+
+func splitPoliciesAndOtherObjects(objects []parser.K8sObject) (policies, nonPolicies []parser.K8sObject) {
+	for _, obj := range objects {
+		switch obj.Kind {
+		case parser.Networkpolicy:
+			policies = append(policies, obj)
+		default:
+			nonPolicies = append(nonPolicies, obj)
+		}
+	}
+	return policies, nonPolicies
+}
+
+// TBD: may be deprcated and use only AddObjectsByKind even when kind is always Networkpolicy?
+func (pe *PolicyEngine) addPolicies(policies []parser.K8sObject) (err error) {
+	for _, policyObj := range policies {
+		err = pe.UpsertObject(policyObj.Networkpolicy)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
this function is indeed not required, can just use `AddObjectsByKind`

Reply:
done